src/Browsershot.php

```diff
@@ -278,6 +278,36 @@ public function setUrl(string $url): static
             }
         }
 
+        $parsedUrl = parse_url($url);
+    
+        // Ensure that the scheme is set and allowed
```

Comment:
Could you split these up in smaller functions with a good name? Eg. `ensureAllowedScheme`, `ensureHostIsPresent`?